doc/Manuals/ug-cs_x-road_6_central_server_user_guide.md

```diff
@@ -1277,71 +1277,87 @@ Prerequisites
 
 1. Shutdown X-Road processes.
 
-    `systemctl stop "xroad*"`
+```bash
+systemctl stop "xroad*"
+```
 
-2. Dump the local database centerui_production to be migrated. The credentials of the database user can be found in `/etc/xroad/db.properties`. Notice that the versions of the local PostgreSQL client and remote PostgreSQL server must match.
+2. Dump the local database centerui_production to be migrated. The credentials of the database admin user can be found in `/etc/xroad.properties`. Notice that the versions of the local PostgreSQL client and remote PostgreSQL server must match.
 
-    `pg_dump -F t -h 127.0.0.1 -p 5432 -U centerui -f centerui_production.dat centerui_production`
+```bash
+pg_dump -F t -h 127.0.0.1 -p 5432 -U centerui_admin -f centerui_production.dat centerui_production
+```
 
 3. Shut down and mask local postgresql so it won't start when xroad-proxy starts.
 
-    `systemctl stop postgresql`
-    `systemctl mask postgresql`
+```bash
+systemctl stop postgresql
+```
+
+```bash
+systemctl mask postgresql
+```
 
-4. Connect to the remote database server as the superuser postgres and create roles, databases and access permissions as follows. Note that the line `GRANT centerui_production to postgres` is AWS RDS specific and not necessary if the postgres user is a true super-user.
+4. Connect to the remote database server as the superuser postgres and create roles, databases and access permissions as follows.
 
-Central Server version 6.23.x
 ```bash
     psql -h <remote-db-url> -p <remote-db-port> -U postgres
     CREATE DATABASE centerui_production ENCODING 'UTF8';
-    CREATE ROLE centerui LOGIN PASSWORD '<centerui-password>';
-    GRANT centerui to postgres;
+    REVOKE ALL ON DATABASE centerui_production FROM PUBLIC;
+    CREATE ROLE centerui_admin LOGIN PASSWORD '<centerui_admin password>';
+    GRANT centerui_admin TO postgres;
+    GRANT CREATE,TEMPORARY,CONNECT ON DATABASE centerui_production TO centerui_admin;
     \c centerui_production
     CREATE EXTENSION hstore;
-```
-
-Additionally for Central Server version 6.24.x or greater
-```sql
-    CREATE SCHEMA centerui AUTHORIZATION centerui;
+    CREATE SCHEMA centerui AUTHORIZATION centerui_admin;
+    REVOKE ALL ON SCHEMA public FROM PUBLIC;
+    GRANT USAGE ON SCHEMA public TO centerui_admin;
+    CREATE ROLE centerui LOGIN PASSWORD '<centerui password>';
+    GRANT centerui TO postgres;
+    GRANT TEMPORARY,CONNECT ON DATABASE centerui_production TO centerui;
+    GRANT USAGE ON SCHEMA public TO centerui;
+    GRANT USAGE ON SCHEMA centerui TO centerui;
+    GRANT SELECT,UPDATE,INSERT,DELETE ON ALL TABLES IN SCHEMA centerui TO centerui;
+    GRANT SELECT,UPDATE ON ALL SEQUENCES IN SCHEMA centerui TO centerui;
+    GRANT EXECUTE ON ALL FUNCTIONS IN SCHEMA centerui to centerui;
```

Comment:
Use generic inputs like '`<database user>`' instead of 'centerui' as there's possibility that client uses customised database users and would have different in db.properties file. Also the same goes for admin user, database name and schema